packages/core/src/providers/blockNumber/blockNumber/provider.tsx

```diff
@@ -14,8 +15,11 @@ interface Props {
 export function BlockNumberProvider({ children }: Props) {
   const { library, chainId } = useEthers()
   const [state, dispatch] = useReducer(blockNumberReducer, {})
+  const { isActive } = useWindow()
 
-  useEffect(() => subscribeToNewBlock(library, chainId, dispatch), [library, chainId])
+  useEffect(() => {
+    subscribeToNewBlock(library, chainId, dispatch, isActive)
```

Comment:
This changed the behaviour!

```suggestion
    return subscribeToNewBlock(library, chainId, dispatch, isActive)
```

Reply:
My mistake with adding these brackets